lib/Command/CreateMultiUser.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * @author Tom Needham <tom@owncloud.com>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Testing\Command;
+
+use OC\Core\Command\Base;
+use OCP\ILogger;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ *
+ * @author Tom Needham <tom@owncloud.com>
+ * Creates, provisions and logs in/out database users.
+ *
+ */
+class CreateMultiUser extends Base {
+
+	/**
+	 * @var IUserManager
+	 */
+	protected $userManager;
+	/**
+	 * @var IUserSession
+	 */
+	protected $session;
+	/**
+	 * @var ILogger
+	 */
+	protected $logger;
+
+	/**
+	 *
+	 * @param IUserManager $userManager
+	 * @param IUserSession $session
+	 * @param ILogger $logger
+	 *
+	 * @return void
+	 */
+	public function __construct(
+		IUserManager $userManager, IUserSession $session, ILogger $logger
+	) {
+		$this->userManager = $userManager;
+		$this->session = $session;
+		$this->logger = $logger;
+		parent::__construct();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see \OC\Core\Command\Base::configure()
+	 *
+	 * @return void
+	 */
+	protected function configure() {
+		parent::configure();
+
+		$defaultUidPrefix = 'zombie-' . \time() . '-';
+
+		$this
+			->setName('testing:createusers')
+			->setDescription('Creates and provisions multiple users for testing')
+			->addOption(
+				'prefix', 'p',
+				InputOption::VALUE_REQUIRED, 'userid prefix for created users',
+				$defaultUidPrefix
+			)
+			->addArgument('numUsers', InputArgument::REQUIRED);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see \Symfony\Component\Console\Command\Command::execute()
+	 *
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 *
+	 * @return int|null null or 0 if everything went fine, or an error code
+	 */
+	protected function execute(
+		InputInterface $input, OutputInterface $output
+	) {
+		$num = $input->getArgument('numUsers');
+		$prefix = $input->getOption('prefix');
+		$progress = new ProgressBar($output, $num);
```

Comment:
need to type hint `$num` to `int` it seems to make phpstan happy

Reply:
strange its happy locally and also casting it to `int` does not help

Reply:
directly casting the output of `getArgument` to int did not help phpstan

checking if its an array and if not cast to int did make it happy
```
		$numUsersArgument = $input->getArgument('numUsers');
		if (\is_array($numUsersArgument)) {
			throw new \Exception("invalid argument");
		} else {
			$num = (int) $numUsersArgument;
		}
```